lib/shape.ml

```diff
@@ -43,6 +50,11 @@ let ellipse ?x ?y rx ry =
   | Some x, Some y -> Ellipse {c = {x; y}; rx; ry}
   | _ -> Ellipse {c = { x = canvas_mid.x; y = canvas_mid.y}; rx; ry}
 
+let line ?x1 ?y1 x2 y2 =
+  match (x1, y1) with 
+  | Some x, Some y -> Line {a = {x;y}; b = {x = x2; y = y2}}
+  | _ -> Line {a = canvas_mid; b = {x = x2; y = y2}}
```

Comment:
To me, this seems like reasonable defaults. Any thoughts @nikochiko?

Reply:
Yeah, this looks OK to me.

One related concern I have is about x1 and y1 being optional. The argument is ignored if only one of the two is passed. This isn't very intuitive. But we are currently doing the same for all shapes, so we should perhaps discuss this in a separate issue and not in the scope of this PR.